simpeg/directives/_regularization.py

```diff
@@ -0,0 +1,482 @@
+        Factor to cool the IRLS threshold epsilon.
+    f_min_change: float
+        Minimum change in the regularization function to continue the IRLS iterations.
+    max_irls_iterations: int
+        Maximum number of IRLS iterations.
+    percentile: float
+        Percentile of the function values used to determine the initial IRLS threshold.
+    """
+
+    def __init__(
+        self,
+        beta_schedule: BetaSchedule | None = None,
+        chifact_start: float = 1.0,
+        chifact_target: float = 1.0,
+        cooling_factor: float = 1.2,
+        f_min_change: float = 1e-2,
+        max_irls_iterations: int = 20,
+        misfit_tolerance: float = 1e-1,
+        percentile: int = 100,
+        **kwargs,
+    ):
+        self._metrics: IRLSMetrics | None = None
+        self.beta_schedule: BetaSchedule = beta_schedule
+        self.chifact_start: float = chifact_start
+        self.chifact_target: float = chifact_target
+        self.cooling_factor: float = cooling_factor
+        self.f_min_change: float = f_min_change
+        self.max_irls_iterations: int = max_irls_iterations
+        self.misfit_tolerance: float = misfit_tolerance
+        self.percentile: int = percentile
+
+        if "silent" in kwargs:
+            warnings.warn(
+                "The `silent` keyword argument is deprecated. "
+                "Use `verbose` instead.",
+                FutureWarning,
+                stacklevel=2,
+            )
+            kwargs["verbose"] = not kwargs.pop("silent")
+
+        super().__init__(**kwargs)
+
+    @property
+    def metrics(self) -> IRLSMetrics:
+        """Various metrics used by the IRLS algorithm."""
+        if self._metrics is None:
+            self._metrics = IRLSMetrics(
+                input_norms=self.get_input_norms(),
+            )
+        return self._metrics
+
+    def get_input_norms(self) -> list[float]:
+        input_norms = []
+        for reg in self.reg.objfcts:
+            if not isinstance(reg, Sparse):
+                continue
+            input_norms += [reg.norms]
+
+        return input_norms
+
+    @property
+    def max_irls_iterations(self) -> int:
+        """Maximum irls iterations."""
+        return self._max_irls_iterations
+
+    @max_irls_iterations.setter
+    def max_irls_iterations(self, value):
+        self._max_irls_iterations = validate_integer(
+            "max_irls_iterations", value, min_val=0
+        )
+
+    @property
+    def misfit_tolerance(self) -> float:
+        """Tolerance on deviation from the target chi factor, as a fractional percent."""
+        return self._misfit_tolerance
+
+    @misfit_tolerance.setter
+    def misfit_tolerance(self, value):
+        self._misfit_tolerance = validate_integer("misfit_tolerance", value, min_val=0)
+
+    beta_tol = deprecate_property(
+        misfit_tolerance,
+        "beta_tol",
+        "UpdateIRLS.misfit_tolerance",
+        future_warn=True,
+        removal_version="0.22.0",
+    )
```

Comment:
Since this is a new class, I don't think we should worry about maintaining deprecated properties from the old class (`Update_IRLS`). Let's just move forward with the properties we want this class to have and how we want it to behave. If anyone needs backward compatibility, we'll still provide the `Update_IRLS` class for a while.
```suggestion
```